api/src/main/java/org/openmrs/api/context/Daemon.java

```diff
@@ -178,7 +178,7 @@ public void run() {
 				isDaemonThread.set(true);
 				try {
 					Context.openSession();
-					runnable.run();
+                    new Thread(runnable).start();
```

Comment:
This is actually intended as you can see the start method is called below in line 189: thread.start(). We need to annotate the run method (line 177) with: 
{code}
@SuppressWarnings("squid:3AS1217") //We intentionally do not start a new thread yet, rather wrap the run call in a session.
{code}